src/SIL.Harmony/Db/CrdtDbContextFactory.cs

```diff
@@ -0,0 +1,97 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.ChangeTracking;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+
+namespace SIL.Harmony.Db;
+
+public class CrdtDbContextFactory<TContext>(IDbContextFactory<TContext> dbContextFactory) : ICrdtDbContextFactory
+    where TContext : DbContext, ICrdtDbContext
+{
+    public async Task<ICrdtDbContext> CreateDbContextAsync(CancellationToken cancellationToken = new CancellationToken())
+    {
+        return await dbContextFactory.CreateDbContextAsync(cancellationToken);
+    }
+
+    public ICrdtDbContext CreateDbContext()
+    {
+        return dbContextFactory.CreateDbContext();
+    }
+}
+
+public interface ICrdtDbContextFactory
+{
+    Task<ICrdtDbContext> CreateDbContextAsync(CancellationToken cancellationToken = new CancellationToken());
+    ICrdtDbContext CreateDbContext();
+}
+
+public class CrdtDbContextFactoryNoContextFactory<TContext>(TContext dbContext) : ICrdtDbContextFactory
```

Comment:
Should this be called `NoDisposeFactory`? I'm not sure what the name should be telling me.